ivi-layermanagement-examples/EGLWLMockNavigation/include/Sky.h

```diff
@@ -0,0 +1,53 @@
+/***************************************************************************
+ *
+ * Copyright (C) 2018 Advanced Driver Information Technology Joint Venture GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ****************************************************************************/
+#ifndef _SKY_H
+#define _SKY_H
+
+#include "IRenderable.h"
+#include "IUpdateable.h"
+#include "vec.h"
+#include "ShaderGradient.h"
+
+class ShaderBase;
+
+class Sky : public IRenderable, public IUpdateable
+{
+public:
+    Sky(vec3f position, vec3f size, vec4f color, ShaderGradient* shader, float maxColorVal = 3.0, float colorChangeSpeed = 0.02);
```

Comment:
seems like color range or color component is defined between 0 and 1.0 but here maxColorVal is 3.0. could you clarify...

Reply:
The color range should be clamped to the range [0, 1] by opengl but in my experiments it looks like it's not. If i stop incrementing at 1.0 in Sky::update(), I get too much green at the end.

Reply:
the open gl will clip it, but the calculations you are doing are not linear, you have a factor 1/2 for green  or 1/5 for blue components maybe this is the reason.

Reply:
I've modified the Sky class a bit, but unfortunately I could not figure out, why I get a different picture at the end of the sunrise if I set color values, which are over 1.0 to 1.0 at the end of the computation. If the color values would be clipped to 1.0, the picture should not change.